File: app/src/main/java/vn/loitp/app/activity/customviews/videoview/exoplayer/ExoPlayerActivity3.java
package vn.loitp.app.activity.customviews.videoview.exoplayer;

import android.os.Bundle;

import com.core.base.BaseFontActivity;
import com.core.utilities.LScreenUtil;
import com.google.android.exoplayer2.ui.PlayerView;
import com.views.exo.PlayerManager;

import loitp.basemaster.R;

public class ExoPlayerActivity3 extends BaseFontActivity {
    private PlayerView playerView0;
    private PlayerView playerView1;
    private PlayerManager playerManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        playerView0 = findViewById(R.id.player_view_0);
        playerView1 = findViewById(R.id.player_view_1);
        final String linkPlay = "https://bitmovin-a.akamaihd.net/content/MI201109210084_1/mpds/f08e80da-bf1d-4e3d-8899-f0f6155f6efa.mpd";
        playerManager = new PlayerManager(getActivity());
        playerManager.init(this, playerView0, linkPlay);

        findViewById(R.id.bt_0_1).setOnClickListener(view -> PlayerView.switchTargetView(playerView0.getPlayer(), playerView0, playerView1));
        findViewById(R.id.bt_1_0).setOnClickListener(view -> PlayerView.switchTargetView(playerView1.getPlayer(), playerView1, playerView0));
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return "TAG" + getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_exo_player3;
    }

    @Override
    protected void onResume() {
        super.onResume();
        playerManager.resumeVideo();
    }

    @Override
    public void onPause() {
        super.onPause();
        playerManager.pauseVideo();
    }

    @Override
    public void onDestroy() {
        playerManager.release();
        super.onDestroy();
    }

    @Override
    public void onBackPressed() {
        if (LScreenUtil.INSTANCE.isLandscape(getActivity())) {
            playerManager.toggleFullscreen(getActivity());
        } else {
            super.onBackPressed();
        }
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/videoview/exoplayer/ExoPlayerActivity3.kt
package vn.loitp.app.activity.customviews.videoview.exoplayer

import android.os.Bundle
import com.loitpcore.annotation.IsFullScreen
import com.loitpcore.annotation.LogTag
import com.loitpcore.core.base.BaseFontActivity
import com.loitpcore.core.utilities.LScreenUtil
import com.google.android.exoplayer2.ui.PlayerView
import com.loitpcore.views.exo.PlayerManager
import kotlinx.android.synthetic.main.activity_video_exo_player3.*
import vn.loitp.app.R

@LogTag("ExoPlayerActivity3")
@IsFullScreen(false)
class ExoPlayerActivity3 : BaseFontActivity() {
    private var playerManager: PlayerManager? = null

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_video_exo_player3
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val linkPlay =
            "https://bitmovin-a.akamaihd.net/content/MI201109210084_1/mpds/f08e80da-bf1d-4e3d-8899-f0f6155f6efa.mpd"
        playerManager = PlayerManager(this)
        playerManager?.init(context = this, playerView = playerView0, linkPlay = linkPlay)

        bt0To1.setOnClickListener {
            playerView0.player?.let { p ->
                PlayerView.switchTargetView(
                    p,
                    playerView0,
                    playerView1
                )
            }
        }
        bt1To0.setOnClickListener {
            playerView1.player?.let { p ->
                PlayerView.switchTargetView(
                    p,
                    playerView1,
                    playerView0
                )
            }
        }
    }

    override fun onResume() {
        super.onResume()
        playerManager?.resumeVideo()
    }

    public override fun onPause() {
        super.onPause()
        playerManager?.pauseVideo()
    }

    public override fun onDestroy() {
        playerManager?.release()
        super.onDestroy()
    }

    override fun onBackPressed() {
        if (LScreenUtil.isLandscape()) {
            playerManager?.toggleFullscreen(activity = this)
        } else {
            super.onBackPressed()
        }
    }
}
